### DISCUSSION TITLE
tspl viewer

### DISCUSSION THREAD

**Discussion by ilia2s** [php-aidc/label-printer#3] [Ideas]
Hello, Can you develop or suggest some TSPL viewer / render software to convert a TSPL commands to bitmap image?

**Comment by jhaoda** [comment]
Hello. Sorry, but i can’t suggest such programs and don’t have time to develop them.

**Reply by xavehoo** [reply]
Hi @ilia2s , there is already a TSPL/TSPL2 command ('PUTBMP') to print BMP format images! :)

**Reply by ilia2s** [reply]
Hello. Sorry, I suppose we talk about a different problems. I need a solution to debug a sticker code that I wrote in TSPL. Now I use a full roll  to create a great sticker's command set when our company need to add a new logistic provider. In this case a tool that can render an image from TSPL will be great. I found this tools for another languages such tsc and etc, but no one of them is not support TSPL. Learning a tons of websites gives me nothing results and better TSPL render that I found is printer only. 

TSPL command PUTBMP it is  just a command to print a bitmap data in the sticker.

**Reply by xavehoo** [reply]
Aah ok, now it's much clearer. Interesting, as soon as I have some time I will think if something similar could be created or added to this project (for debugging purpose) with little effort or not.

**Reply by siremarcus** [reply]
Hello @xavehoo ! did you have time for this request ? ;) 
I did not see references to it in the readme
